src/setup/cortx_deploy

```diff
@@ -110,27 +133,57 @@ node_id=`cat /etc/machine-id`
-cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
-
-# Apply Common Config
-log $INFO "cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE' $FORCE_PROVISIONING"
-cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
-
-# Validate config
-if ! [ -z $VALIDATE_CONF ];
-then
-    log $INFO "cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c '$CORTX_CONFSTORE'"
-    cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE" || exit 1
-    log $INFO "cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE'"
-    cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE" || exit 1
+# Check current image version
+ installed_version=$(conf_get $CORTX_CONFSTORE "cortx>common>release>version")
+# if current version is not available in confstore
+if [ $ installed_version = "[null]" ]; then
+    ACTION="deploy"
+else
+    # Below can be used when subversioning is considered.
+    #current_prim_version=$(awk -F. '{print $1 "." $2 "." $3}' <<< $ installed_version)
+    current_version=$(awk -F. '{print $4}' <<< $ installed_version)
 fi
 
-# Bootstrap Cluster
-log $INFO "cortx_setup cluster bootstrap -c '$CORTX_CONFSTORE' $FORCE_PROVISIONING"
-cortx_setup cluster bootstrap -c "$CORTX_CONFSTORE" $FORCE_PROVISIONING;
+image_version=$(sed -e 's/[][]//g' -e 's/"//g' <<< $(conf_get $RELEASE_INFO "VERSION"))
+# Below can be used when subversioning is considered.
+#update_prim_version=$(awk -F- '{print $1}' <<<  $image_version )
+update_version=$(awk -F- '{print $2}' <<<  $image_version)
+
+if [ $current_version -eq $update_version ]; then
+    if [ ! -z $FORCE_PROVISIONING ]; then
+        ACTION="deploy"
+    else
+        return 0
+    fi
+elif [ $current_version -gt $update_version ]; then
+    log $ERROR "Cannot update Cortx Image current_version ($installed_version) > update_version ($image_version)"
+    exit 1
+else
+    ACTION="upgrade"
+fi
 
+case $ACTION in
+        deploy)
+            apply_cluster_config
+            apply_common_config
+            # Validate config
+            if ! [ -z $VALIDATE_CONF ]; then
+                log $INFO "cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c '$CORTX_CONFSTORE'"
+                cortx_setup config validate -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE" || exit 1
+                log $INFO "cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE'"
+                cortx_setup config validate -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE" || exit 1
+            fi
+            # Deploy cortx
+            deploy
+            ;;
+        upgrade)
+            # Upgrade cortx
+            upgrade
+            ;;
+        *)
+            log $ERROR "No action selected, check Cortx current_version ($installed_version) and update_version ($image_version)"
```

Comment:
This error message does not help user in anyway. We need to be more meaningful. 

log $ERROR "Internal error. Could not determine version. Invalid image. Installed version is $current_version, Image version is $image_version."

Reply:
Done